test/date.js

```diff
@@ -304,6 +316,83 @@ describe('date', () => {
             });
         });
 
+        describe('timestamp()', () => {
+
+            it('validates javascript timestamp', (done) => {
+
+                const now = new Date();
+                const milliseconds = now.getTime();
+
+                Joi.date().timestamp().validate(milliseconds, (err, value) => {
+
+                    expect(err).to.not.exist();
+                    expect(value).to.deep.equal(now);
+                });
+                Joi.date().timestamp('javascript').validate(milliseconds, (err, value) => {
+
+                    expect(err).to.not.exist();
+                    expect(value).to.deep.equal(now);
+                });
+                done();
+            });
+
+            it('validates unix timestamp', (done) => {
+
+                const now = new Date();
+                const seconds = now.getTime() / 1000;
+
+                Joi.date().timestamp('unix').validate(seconds, (err, value) => {
+
+                    expect(err).to.not.exist();
+                    expect(value).to.deep.equal(now);
+                    done();
+                });
+            });
+
+            it('validates timestamps with decimals', (done) => {
+
+                Helper.validate(Joi.date().timestamp(), [
+                    [new Date().getTime().toFixed(4), true]
+                ]);
+                Helper.validate(Joi.date().timestamp('javascript'), [
+                    [new Date().getTime().toFixed(4), true]
+                ]);
+                Helper.validate(Joi.date().timestamp('unix'), [
+                    [(new Date().getTime() / 1000).toFixed(4), true]
+                ]);
+                done();
+            });
+
+            it('fails with not allowed type', (done) => {
+
+                expect(() => {
+
+                    Joi.date().timestamp('not allowed');
+                }).to.throw(Error, new RegExp('"type" must be one of'));
+                done();
+            });
+
+            it('validates timestamps with a friendly error message', (done) => {
```

Comment:
Not sure this test is needed once previous tests also validate the error message.